### ISSUE TITLE
2022/03/06

### ISSUE THREAD

**Issue by nahyeon00** [NLL-JBNU/Nahyeon#3]
# 논문 제목 : Learning to Classify Open Intent via Soft Labeling and Manifold Mixup
### 논문 링크: [Learning to Classify Open Intent via Soft Labeling and Manifold Mixup](https://ieeexplore.ieee.org/document/9693239)

* 이번 주 목표
    * 'Learning to Classify Open Intent via Soft Labeling and Manifold Mixup' 논문 발표 준비
    * oos dataset 계층적 label로 분류하고 confusion matrix로 예측값 확인해보기
   
< __논문 발표 요약__>
  
* Open intent classification
    * objective : open intent 를 찾는 동시에 known intent 를 정확하게 분류하는 것

* Existing methods
    *  대부분 k-class classifier 와 합쳐진 outlier detection algorithm 사용
    * open intent classification 위해 (k+1)-classifier 직접 학습
        *  k-class 샘플을 (k+1)-class classifier 로 training 시키기 어려움 존재

* Two main challenges 해결 방법
    * known class 의 boundary 보정 위해 soft labeling을 이용하여 intent label 분포를 재구성하여 해결
    * test 샘플을 k+1 번째 open intent class로 boundary 학습을 위해 manifold mixup을 이용하여 open intent에 대한 pseudo sample 생성
 
* __The model architecture__ 

![image](https://user-images.githubusercontent.com/90039228/156923752-916dc045-e639-411c-af8f-98307b79e054.png)

    * Deep intent classificaiton model & Pre-training for known intents
        * BERT 를 사용하여 intent representation 추출
        * mean pooling, dense layer(ReLU) 사용
        * softmax loss를 이용하여 더 좋은 intent representation 을 얻기 위해 pretrain
        
    * Training for open intent (soft labeling & manifold mixup)
        * open intent data 부족으로 open intent 구별 위한 model train에 어려움 존재
        * training set 에서 label 분포를 reshape 하므로서 known intent 에 대한 pseudo data 생성을 위해 soft labeling 사용
             * kl-divergence loss 로 학습 
        *  test 샘플을 k+1 번째 open intent class로 boundary 학습을 위해 manifold mixup을 이용하여 open intent에 대한 pseudo sample 생성
            * softmax lass 로 학습
        * soft labeling loss 와 manifold mixup loss 합으로 전체 training
  
< 현재 실험 진행 상황 >
* 8개의 계층적 label 로 임의 분류
    * pto, 대답, 돈, 예약식당, 리스트날짜, 음식, 노래 계열
* confusion matrix로 label마다  예측한 값 확인

* set
    * model: multi-base-multilingual-cased
    * data: oos(영어 기존 버전
    * known_cls_ratio: 0.75
 
* 다음주까지 해야할 것
    * 'Manifold Mixup: Better REpresentations by Interpolating Hidden States' 논문 다시 읽어보기
    * 실험에서 나아갈 부분 자료 확인하여 고민하기
    * 이번 논문 'Learning to Classify Open Intent via Soft Labeling and Manifold Mixup' 코드를 oos 데이터로 돌려서 결과 확인해보기
